[Jolyn Loh] iP

jolynloh opened the pull request:
### Let Skyler help you manage tasks, deadlines and more! 
> "Edit your life frequently and ruthlessly. It’s your masterpiece after all.” – Nathan W. Morris ([source](https://dansilvestre.com/productivity-quotes/#12_Edit_your_life_frequently_and_ruthlessly_Its_your_masterpiece_after_all_-_Nathan_W_Morris))

Skyler :robot: , your friendly personal chat assistant, is here to free your mind of having to remember things you need to do. It's

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download the `.jar` file from [here](https://github.com/jolynloh/ip/releases/download/A-Jar/skyler.jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you

And it is __FREE__!

Features:

- [x] Managing tasks
- [ ] GUI component

***

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public static void main(String[] args) {
        new Skyler("data/skyler.txt").run();
    }
```

Commit:
Add Gradle support

jolynloh committed:
Level 1: Greet, Echo, Exit

jolynloh committed:
Level 2: Add, List

jolynloh committed:
Level 3: Mark as Done

jolynloh committed:
Fix: Tell don't ask

jolynloh committed:
Level 4: ToDos, Events, Deadlines

jolynloh committed:
A-TextUiTesting: Automated Text UI Testing

jolynloh committed:
Level 5: Handle Errors

jolynloh committed:
Level 6: Delete

jolynloh committed:
Add save task feature

jolynloh committed:
Add date-time formatting

jolynloh committed:
Add save task feature

jolynloh committed:
Add date-time formatting

jolynloh committed:
Refactor code to align more with OOP

jolynloh committed:
Organise classes into java package

jolynloh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

jolynloh committed:
Automate project builds using Gradle

jolynloh committed:
Add JUnit tests

jolynloh committed:
Add JavaDoc comments

jolynloh committed:
Edit code to comply with coding standard

jolynloh committed:
Add task search feature

jolynloh committed:
Merge branch 'branch-A-JavaDoc'

jolynloh committed:
Merge branch 'branch-A-CodingStandard'

jolynloh committed:
Merge branch 'branch-Level-9'

teekaytai commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
+<<<<<<< HEAD
```
Oops, looks like a git conflict marker was left here by mistake

teekaytai commented on src/main/java/skyler/Storage.java:
```diff
@@ -0,0 +1,232 @@
+        String filePath = new File(currPath).getAbsolutePath();
+        File directory = new File(filePath);
+
+        String[] flist = directory.list();
```
Perhaps this variable name could avoid abbreviating the f to make it easier to understand?

teekaytai commented on src/main/java/skyler/Event.java:
```diff
@@ -0,0 +1,45 @@
+ */
+public class Event extends Task {
+
+    protected LocalDateTime at;
```
Maybe this variable name could be changed to make its purpose more clear without needing to see the context?

teekaytai commented on src/main/java/skyler/Parser.java:
```diff
@@ -0,0 +1,77 @@
+            taskList.list();
+            return 1;
+        } else if (command.startsWith("mark")) {
+            int item = Integer.parseInt(command.substring(command.length() - 1));
```
What if this name was changed to `itemId` instead to help show that it's a number?

teekaytai commented on src/main/java/skyler/Storage.java:
```diff
@@ -0,0 +1,232 @@
+     * @throws IOException If IO operation fails.
+     */
+    public void saveTask(ArrayList<Task> task) throws IOException {
+        String filepath = new File(filePathChosen).getAbsolutePath();
```
Just a nitpick but perhaps this variable could be renamed to `filePath` to keep it consistent with other `filePath` variables in the code

teekaytai commented on src/main/java/skyler/Parser.java:
```diff
@@ -0,0 +1,77 @@
+            taskList.unmark(item);
+            return 1;
+        } else if (command.startsWith("todo")) {
+            String[] arr = command.split(" ", 2);
```
Maybe a name like `commandArguments` could help explain what this array contains?

jolynloh replied:
Thank you for the suggestions! 😄

teekaytai reviewed (commented):
Overall the code is very readable and easy to understand. Just a few nitpicks and possible suggestions for changes

ish1506 commented on src/main/java/skyler/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task that needs to be done before a specific date and time
```
Don't forget to include periods in your Javadoc comments! I noticed the same issue in other places too.

ish1506 commented on src/main/java/skyler/Storage.java:
```diff
@@ -0,0 +1,232 @@
+            char type = taskStr.charAt(0);
+            int taskStatus = Character.getNumericValue(taskStr.charAt(4));
+
+            switch(type) {
```
Missing a whitespace here, easily fixed with the IDE configurations.
```suggestion
            switch (type) {
```

ish1506 commented on src/main/java/skyler/Task.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
```
Minor issue but I think the brackets can be removed.
```suggestion
        return isDone ? "X" : " ";
```

ish1506 commented on src/main/java/skyler/TaskList.java:
```diff
@@ -0,0 +1,228 @@
+    /**
+     * Prints the current list of tasks
+     */
+    public void list() {
```
Perhaps `printList` is a clearer method name?

ish1506 commented on src/main/java/skyler/TaskList.java:
```diff
@@ -0,0 +1,228 @@
+     *
+     * @param item Index of task.
+     */
+    public void delete(int item) {
```
Perhaps `index` or `itemIndex` are clearer names?

jolynloh replied:
Thanks for the suggestions! 😃

ish1506 reviewed (commented):
LGTM, only some nits here and there. Well done!

jolynloh committed:
Add GUI using JavaFX

jolynloh committed:
Reformat code to use FXML

jolynloh committed:
Update GUI to be fully-functional and responsive

jolynloh committed:
Merge branch 'branch-Level-10'

jolynloh committed:
Enhance GUI to be more user-friendly

jolynloh committed:
Add assertions

Program does not document important assumptions that should hold at
various points in the code.

Assertions allow runtime to verify assumptions about the program.
Assertion failures help detect errors in the program.

Let's update the code to include assertion statements.

jolynloh committed:
Update assertions

jolynloh committed:
Improve code quality

Code readability is poor with long methods and several violations of
SLAP. Multiple uses of magic literals hinder understanding of code.

Unifying duplicated code and using meaninful names improve code quality.

Let's
* extract methods
* use named constants that tell us more about the meaning of a number or
  string

jolynloh committed:
Merge pull request #1 from jolynloh/branch-A-Assertions

Add assertions

jolynloh committed:
Fix merge conflicts

jolynloh committed:
Merge pull request #3 from jolynloh/branch-A-CodeQuality

Improve code quality

jolynloh committed:
Add reschedule task feature

jolynloh committed:
Merge branch 'branch-B-Snooze'

jolynloh committed:
Improve GUI

jolynloh committed:
Add a User Guide

jolynloh committed:
Update User Guide

jolynloh committed:
Update README
